test/CMakeLists.txt

```diff
@@ -21,7 +21,13 @@ add_executable(
   test_eos_vinet.cpp
   test_eos_noble_abel.cpp
   test_eos_stiff.cpp
-)
+  test_eos_helmholtz.cpp
+  )
+
+if(SINGULARITY_TEST_HELMHOLTZ)
+    configure_file(${PROJECT_SOURCE_DIR}/data/helmholtz/helm_table.dat ${CMAKE_BINARY_DIR}/data/helmholtz/helm_table.dat COPYONLY)
+  target_compile_definitions(eos_unit_tests PRIVATE SINGULARITY_TEST_HELMHOLTZ SINGULARITY_USE_HELMHOLTZ)
+endif()
```

Comment:
I see what's happening here, and don't object to holding up a merge, but this should be returned to.

`test_eos_helmholtz.cpp` should only be included in the `eos_unit_tests` source list if `_USE_HELMOLTZ` is on. The compile def here is unnecessary (aside from the conditional compilation of `test_eos_helmholtz.cpp`, which previous would make redundant). 

`_USE_HELMOLTZ` should be included in the target interface of `singularity-eos`, e.g.

```cmake
# defines
if(SINGULARITY_USE_SINGLE_LOGS)
  target_compile_definitions(singularity-eos PUBLIC SINGULARITY_USE_SINGLE_LOGS)
endif()
``` 

Relatedly, the tests in `test_eos_unit.cpp` related to `_TEST_SESAME` and `_TEST_EOSPAC` should be moved into separate source files.

Reply:
To clarify, the `_USE_HELMHOLTZ` compile option _should_ be changed (downstream codes will need to know that `_USE_HELMHOLTZ` is needed). Other issues can wait.

Reply:
Ah wait, I got mixed up. Sorry - it is present as it should be. The second inclusion in the tests is what threw me.